kv_test.go

```diff
@@ -0,0 +1,438 @@
+package tiledb
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// ExampleKV example of creating and using a Key-Value store
+func ExampleKV() {
+
+	// Create Context with default configuration
+	context, err := NewContext(nil)
+
+	if err != nil {
+		// handle error
+		return
+	}
+
+	// Create a KV schema
+	kvSchema, err := NewKVSchema(context)
+	if err != nil {
+		// handle error
+		return
+	}
+
+	// Add a attribute to the key value store
+	a1, err := NewAttribute(context, "a1", TILEDB_INT8)
+	if err != nil {
+		// handle error
+		return
+	}
+
+	err = kvSchema.AddAttributes(a1)
+	if err != nil {
+		// handle error
+		return
+	}
+
+	// create temp KV folder
+	tmpKVPath := os.TempDir() + string(os.PathSeparator) + "tiledb_test_kv"
+	defer os.RemoveAll(tmpKVPath)
+	if _, err := os.Stat(tmpKVPath); err == nil {
+		os.RemoveAll(tmpKVPath)
+	}
+
+	// Create new KV Struct
+	kv, err := NewKV(context, tmpKVPath)
+	if err != nil {
+		// handle error
+		return
+	}
+
+	// Create new KV
+	err = kv.Create(kvSchema)
+	if err != nil {
+		// handle error
+		return
+	}
+
+	// Create a new KVItem to add
+	kvItem, err := NewKVItem(context)
+	if err != nil {
+		// handle error
+		return
+	}
+
+	err = kvItem.SetKey("string_key")
+	if err != nil {
+		// handle error
+		return
+	}
+	err = kvItem.SetValue("a1", int8(1))
+	if err != nil {
+		// handle error
+		return
+	}
+
+	// Add the item to the kv store
+	err = kv.AddItem(kvItem)
```

Comment:
I do not think you are opening the kv in write mode by default?

Reply:
You are correct, I've adjust the example to properly open the kv in write mode before writing. I'm surprised I did not get a tiledb error about not having it opened in write mode?